### PR TITLE
tcurl as http & test

### PR DESCRIPTION
**Pull request by ShanniLi**
@Raynos @anson627

### CODE REVIEW

**Comment by Raynos** [comment]
Why go through proxy ?

**Comment by ShanniLi** [comment]
proxy removed. :)

File: as-http.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    self.subClient.waitForIdentified(
+        {host: self.remoteHostPort},
+        function onIdentified() {
+            self.asHttpClient.sendRequest(
+                req,
+                hreq,
+                function sent(err, head, body) {
+                    var str;
+                    if (err) {
+                        self.logger.log('error', err);
```
Comment (anson627) [inline as-http.js]: why not use logtron? I guess this is more a question instead of comment.

**Comment by anson627** [comment]
lgtm

File: as-http.js (hunk 1 above)
Comment (ShanniLi) [inline as-http.js]: It was started with this simple logger. I never think about moving to logtron. I guess it is a good idea to use logtron. I will do it in next PR. :)

**Comment by Raynos** [comment]
lgtm

File: index.js [hunk 2]
```diff
@@ -175,6 +179,18 @@ function tcurl(opts) {
...
             request.send(opts.endpoint, opts.head, opts.body,
                 onResponse);
+        } else if (opts.http) {
+            var ashttp = TCurlAsHttp({
+                remoteHostPort: opts.hostname + ':' + opts.port,
+                serviceName: opts.service,
+                method: opts.http,
+                path: opts.endpoint,
+                headers: JSON.parse(opts.head),
+                body: JSON.parse(opts.body),
```
Comment (junchaowu) [inline index.js]: if this is http, why does it use json to parse html body?

Comment (Raynos) [inline index.js]: ;) ;) its http+json

Comment (anson627) [inline index.js]: http is a way of transport, json is a way of encoding.

Comment (junchaowu) [inline index.js]: it really depends. Better document it.

Comment (ShanniLi) [inline index.js]: opts.head and opts.body are from command line options. JSON is just a way to allow specify cmd line inputs as objects.

**Pull request merged**